app/models/reservation.rb

```diff
@@ -0,0 +1,6 @@
+class Reservation < ApplicationRecord
+  belongs_to :property
+  belongs_to :user
+
+  enum status: %w(pending approved completed cancelled)
```

Comment:
I like the use of an Enum here. I prefer this over the idea of doing some sort of status_reservations join table so that a reservation can have more than one status (ie, "approve", "completed"). A reservation should only have 1 state for the purposes of this project as you did. Good shit 👍